src/org/omegat/core/dictionaries/LingvoDSL.java

```diff
@@ -61,48 +67,56 @@ public IDictionary loadDict(File file) throws Exception {
 
     @Override
     public IDictionary loadDict(File file, Language language) throws Exception {
-        return new LingvoDSLDict(loadData(file, language));
-    }
-
-    private static DictionaryData<String> loadData(File file, Language language) throws Exception {
-        DictionaryData<String> data = new DictionaryData<>(language);
-        StringBuilder word = new StringBuilder();
-        StringBuilder trans = new StringBuilder();
-        Files.lines(file.toPath(), StandardCharsets.UTF_16).filter(line -> !line.isEmpty() && !line.startsWith("#"))
-                .map(line -> RE_SKIP.matcher(line).replaceAll("")).forEach(line -> {
-                    if (Character.isWhitespace(line.codePointAt(0))) {
-                        trans.append(line.trim()).append('\n');
-                    } else {
-                        if (word.length() > 0) {
-                            data.add(word.toString(), trans.toString());
-                            word.setLength(0);
-                            trans.setLength(0);
-                        }
-                        word.append(line);
-                    }
-                });
-        if (word.length() > 0) {
-            data.add(word.toString(), trans.toString());
-        }
-        data.done();
-        return data;
+        return new LingvoDSLDict(file, language);
     }
 
     static class LingvoDSLDict implements IDictionary {
         protected final DictionaryData<String> data;
+        private final String dataFile;
+
+        LingvoDSLDict(File file, Language language) throws Exception {
+            dataFile = file.getPath();
+            data = new DictionaryData<>(language);
+            if (dataFile.endsWith(".dz")) {
+                try (InputStreamReader reader = new InputStreamReader(new GZIPInputStream(new FileInputStream(file),
+                        8192), StandardCharsets.UTF_16)) {
+                    loadData((new BufferedReader(reader)).lines());
```

Comment:
`BufferedReader` could be included in the `try (...)` statement so that it gets auto-closed.

(The implementation on Java 8 just closes the reader that it wraps, so semantically there's no difference, but it feels nicer to close it.)

In case you put `BufferedReader` outside the `try (...)` because the line was getting too long, I should point out that you can declare multiple variables inside the `try` and they will all get closed ([details](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)):

```suggestion
                try (FileInputStream fis = new FileInputStream(file);
                        GZIPInputStream gis = new GZIPInputStream(fis, 8192);
                        InputStreamReader isr = new InputStreamReader(gis,  StandardCharsets.UTF_16);
                        BufferedReader reader = new BufferedReader(isr)) {
                    loadData(reader.lines());
```